hazelcast-client-legacy/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -401,9 +401,9 @@ protected Object putInternal(final K key, final V value, final ExpiryPolicy expi
             }
         } else {
             if (nearCache != null || statisticsEnabled) {
-                future.andThen(new ExecutionCallback<Object>() {
```

Comment:
Could it be a solution to use same thread for executor service? Something like this:

```
// executes tasks in current thread.
public class SameThreadExecutor implements Executor {
   public void execute(Runnable task) 
}

future.andThen(new ExecutionCallback(), sameThreadExecutor);